src/Registry/TypeRegistry.php

```diff
@@ -920,6 +999,8 @@ protected function prepare_field( string $field_name, array $field_config, strin
 			$field_config['name'] = lcfirst( $field_name );
 		}
 
+		$field_config = self::prepare_config_for_introspection( $field_config );
```

Comment:
We should be able to do this _at least_ as late after the `if ( is_array( $field_config['type'] ) ) {}` block (L1067) in this commit)